test/e2e/l7networkpolicy_test.go

```diff
@@ -333,3 +325,95 @@ func testL7NetworkPolicyTLS(t *testing.T, data *TestData) {
+	l7ProtocolAllowsPathHostname := []crdv1beta1.L7Protocol{
+		{
+			HTTP: &crdv1beta1.HTTPProtocol{
+				Method: "GET",
+				Path:   "/host*",
+			},
+		},
+	}
+	// Create one L7 NetworkPolicy that allows HTTP path 'hostname', and probe twice
+	// where HTTP path 'hostname' is allowed yet 'clientip' will be rejected.
+	createL7NetworkPolicy(t, data, true, policyAllowPathHostname, 1, clientPodLabels, serverPodLabels, ProtocolTCP, p8080, l7ProtocolAllowsPathHostname)
+	time.Sleep(networkPolicyDelay)
+	probeL7NetworkPolicyHTTP(t, data, serverPodName, clientPodName, serverIPs, true, false)
+
+	// Define log matchers for expected L7 NetworkPolicies log entries.
+	var l7LogMatchers []L7LogEntry
+	clientMatcher := L7LogEntry{EventType: "alert", Protocol: "TCP", Http: L7LogHttpEntry{Port: 8080, Url: "/clientip"}}
+	hostMatcher := L7LogEntry{EventType: "http", Protocol: "TCP", Http: L7LogHttpEntry{Port: 8080, Url: "/hostname"}}
+	for _, ip := range serverIPs {
+		clientMatcher.Http.Hostname, hostMatcher.Http.Hostname = ip.String(), ip.String()
+		l7LogMatchers = append(l7LogMatchers, clientMatcher, hostMatcher)
+	}
+
+	checkL7LoggingResult(t, data, l7LoggingNode, l7LogMatchers)
+}
+
+// Partial entries of L7 NetworkPolicy logging necessary for testing.
+type L7LogHttpEntry struct {
+	Hostname string `json:"hostname"`
+	Port     int32  `json:"http_port"`
+	Url      string `json:"url"`
+}
+
+type L7LogEntry struct {
+	EventType string         `json:"event_type"`
+	Protocol  string         `json:"proto"`
+	Http      L7LogHttpEntry `json:"http"`
+}
+
+func checkL7LoggingResult(t *testing.T, data *TestData, nodeName string, matchers []L7LogEntry) {
+	// L7 audit logging filename from generated Suricata config [pkg/agent/controller/networkpolicy/l7engine/reconciler.go:64].
+	l7LogFile := path.Join(logDir, fmt.Sprintf("l7engine/eve-%s.json", time.Now().Format(time.DateOnly)))
+	antreaPodName, err := data.getAntreaPodOnNode(nodeName)
+	require.NoError(t, err, "Error occurred when trying to get the Antrea Agent Pod running on Node %s", nodeName)
+	cmd := []string{"cat", l7LogFile}
+
+	if err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 10*time.Second, false, func(ctx context.Context) (bool, error) {
+		stdout, stderr, err := data.RunCommandFromPod(antreaNamespace, antreaPodName, "antrea-agent", cmd)
+		if err != nil || stderr != "" {
+			// file may not exist yet
+			t.Logf("Error when printing the audit log file, err: %v, stderr: %v", err, stderr)
+			return false, nil
+		}
+		if !strings.Contains(stdout, "http") || !strings.Contains(stdout, "alert") {
+			t.Logf("Audit log file does not contain enough entries yet")
+			return false, nil
+		}
+		var gotLogs []L7LogEntry
+		dec := json.NewDecoder(strings.NewReader(stdout))
+		for {
+			var log L7LogEntry
+			if err := dec.Decode(&log); err == io.EOF {
+				break
+			}
+			require.NoError(t, err)
+			gotLogs = append(gotLogs, log)
+		}
```

Comment:
Because the log entries from `stdout` are a stream line of json objects `{}\n{}\n` instead of a list of json objects. It might be unnecessary to manually modify `stdout` to `[{},{},]` then use unmarshal 🤔

Reply:
Got it, makes sense